packages/loopring_v3/circuit/Gadgets/MathGadgets.h

```diff
@@ -756,6 +945,148 @@ class RequireNotEqualGadget : public GadgetT
     }
 };
 
+// (A <(=) B)
+class LeqGadget : public GadgetT
```

Comment:
In some cases yes, in other cases we need these multiple results for the same comparison, so it's much more efficient to reuse the main comparison.

The highest cost by far is the actual comparison because the cost is dependent on the bit size. So for 96 bit values the cost is already high > 100 constraints, a couple of extra constraints so we don't need to write multiple gadgets for this while making circuits easier to write is a small price to pay. Especially because we don't really use these comparisons often.